front/app/components/AddToBasketButton/useAssignBudget.ts

```diff
@@ -0,0 +1,180 @@
+import { useCallback, useState } from 'react';
+
+// hooks
+import useAuthUser from 'api/me/useAuthUser';
+import useIdeaById from 'api/ideas/useIdeaById';
+import useProjectById from 'api/projects/useProjectById';
+import usePhases from 'api/phases/usePhases';
+import useBasket from 'api/baskets/useBasket';
+import useAddBasket from 'api/baskets/useAddBasket';
+import useBasketsIdeas from 'api/baskets_ideas/useBasketsIdeas';
+import useDeleteBasketsIdea from 'api/baskets_ideas/useDeleteBasketsIdea';
+import useAddBasketsIdea from 'api/baskets_ideas/useAddBasketsIdeas';
+
+// utils
+import { getLatestRelevantParticipationContext } from './utils';
+import { capitalizeParticipationContextType } from 'utils/helperUtils';
+import eventEmitter from 'utils/eventEmitter';
+
+// tracks
+import { trackEventByName } from 'utils/analytics';
+import tracks from 'components/AddToBasketButton/tracks';
+
+// constants
+import { BUDGET_EXCEEDED_ERROR_EVENT } from 'components/ParticipationCTABars/VotingCTABar/events';
+
```

Comment:
Should we move these ERROR_EVENT constants into the votingMethodUtils :thinking: Since they are used in multiple places now?